spec/services/publishers/ats_api/create_vacancy_service_spec.rb

```diff
@@ -41,6 +40,172 @@
+            params[:is_job_share] = true
+            create_vacancy_service
+            expect(Vacancy.last.is_job_share).to be(true)
+          end
+
+          it "gets set as 'true' when provided as a string" do
+            params[:is_job_share] = "true"
+            create_vacancy_service
+            expect(Vacancy.last.is_job_share).to be(true)
+          end
+
+          it "gets set as 'false' when provided as a boolean" do
+            params[:is_job_share] = false
+            create_vacancy_service
+            expect(Vacancy.last.is_job_share).to be(false)
+          end
+
+          it "gets set as 'false' when provided as a string" do
+            params[:is_job_share] = "false"
+            create_vacancy_service
+            expect(Vacancy.last.is_job_share).to be(false)
+          end
+
+          it "gets set as 'false' when any other string come" do
+            params[:is_job_share] = "foobar"
+            create_vacancy_service
+            expect(Vacancy.last.is_job_share).to be(false)
+          end
+        end
+      end
+
+      describe "'visa_sponsorship_available'" do
+        it "defaults to false when not provided" do
+          create_vacancy_service
+          expect(Vacancy.last.visa_sponsorship_available).to be(false)
+        end
+
+        context "when provided" do
+          it "gets set as 'true' when provided as a boolean" do
+            params[:visa_sponsorship_available] = true
+            create_vacancy_service
+            expect(Vacancy.last.visa_sponsorship_available).to be(true)
+          end
+
+          it "gets set as 'true' when provided as a string" do
+            params[:visa_sponsorship_available] = "true"
+            create_vacancy_service
+            expect(Vacancy.last.visa_sponsorship_available).to be(true)
+          end
+
+          it "gets set as 'false' when provided as a boolean" do
+            params[:visa_sponsorship_available] = false
+            create_vacancy_service
+            expect(Vacancy.last.visa_sponsorship_available).to be(false)
+          end
+
+          it "gets set as 'false' when provided as a string" do
+            params[:visa_sponsorship_available] = "false"
+            create_vacancy_service
+            expect(Vacancy.last.visa_sponsorship_available).to be(false)
+          end
+
+          it "gets set as 'false' when any other string come" do
+            params[:visa_sponsorship_available] = "foobar"
+            create_vacancy_service
+            expect(Vacancy.last.visa_sponsorship_available).to be(false)
+          end
+        end
+      end
+
+      describe "'ect_suitable'" do
+        it "defaults ect_status to 'ect_unsuitable when not provided" do
+          create_vacancy_service
+          expect(Vacancy.last.ect_status).to eq("ect_unsuitable")
+        end
+
+        context "when provided" do
+          it "sets 'ect_status' to 'ect_suitable' when provided as a boolean" do
```

Comment:
[minor] "sets 'ect_status' to 'ect_suitable' when params[:ect_suitable] is true (a boolean)" or something like that might be a bit more understandable. This description does not mention that the boolean is true at the moment. Same goes for the other etc_status tests.